core/sdk/src/extn/client/extn_client.rs

```diff
@@ -227,7 +252,45 @@ impl ExtnClient {
                     if message.payload.is_response() {
                         Self::handle_single(message, self.response_processors.clone());
                     } else if message.payload.is_event() {
-                        Self::handle_vec_stream(message, self.event_processors.clone());
+                        //TODO: Need to check if this can be moved to ExtnClient struct itself rather than
+                        //      computing each time
+                        let senders: Vec<CSender<CExtnMessage>> = self
+                            .extn_sender_map
+                            .read()
+                            .unwrap()
+                            .iter()
+                            .map(|(_, v)| v)
+                            .cloned()
+                            .collect();
+                        let is_main = self.sender.get_cap().is_main();
+
+                        if Self::is_ripple_context(&message) {
+                            let mut ripple_context = self.ripple_context.write().unwrap();
+                            if let ExtnPayload::Event(ExtnEvent::DistributorTokenChange(
+                                _dist_token,
```

Comment:
Can we update this `_dist_token` token in Account session?

Reply:
To add on we expect Session Token to be available on updation. Then we can certainly store it in the Context instead of storing just the `DistributorTokenAvailable` enumeration